### PR TITLE
docs: add http-timeouts docs

### PR DESCRIPTION
**Pull request by phantooom**
**What type of PR is this?**

* "docs: add http timeouts docs"

### CODE REVIEW

File: site/content/en/latest/user/http-timeouts.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+__Note:__  The Request duration must be >= BackendRequest duration
+
+## Installation
+
+Follow the steps from the [Quickstart Guide](quickstart.md) to install Envoy Gateway and the example manifest.
+Before proceeding, you should be able to query the example backend using HTTP.
+
+## Verification
+
+Httpbin has the ability to delay responses; we use it as the backend to control response time.
```
Comment (guydc) [inline site/content/en/latest/user/http-timeouts.md:20]: We recently updated the echo-server in use in EG to a version that supports delayed execution, e.g. see here: https://github.com/envoyproxy/gateway/blob/main/site/content/en/latest/user/circuit-breaker.md 

So, you can probably avoid the `httpbin` installation prerequisite and base this demo on the quickstart example, making this a bit simpler for users.

Comment (phantooom) [reply site/content/en/latest/user/http-timeouts.md:20]: 👌

File: site/content/en/latest/user/http-timeouts.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+< HTTP/1.1 504 Gateway Timeout
+< content-length: 24
+< content-type: text/plain
+< date: Fri, 01 Mar 2024 08:43:35 GMT
+<
+* Connection #0 to host 127.0.0.1 left intact
+upstream request timeout
+```
+
+### backendRequest timeout
```
Comment (arkodg) [inline site/content/en/latest/user/http-timeouts.md:119]: lets rm `backendRequest`, its not any extra value imo

File: site/content/en/latest/user/http-timeouts.md (hunk 1 above)
Comment (arkodg) [inline site/content/en/latest/user/http-timeouts.md:20]: using the `/delay/` endpoint

Comment (arkodg) [inline site/content/en/latest/user/http-timeouts.md:20]:
```suggestion
The `httpbin` backend app has the ability to delay responses using the `/delay/` endpoint.
```

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**